### PR TITLE
[Fix](BE) update pid file after be launch succeed

### PR DESCRIPTION
**Pull request by GoGoWen**
# Proposed changes

This pr going to write pid to pid file after be is started.   Before the pid will write to pid file even be is start failed,  which will cause pid to be overwrite by run "sh start_be.sh --daemon" when the be is already started.

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/service/doris_main.cpp [hunk 1]
```diff
@@ -488,6 +465,28 @@ int main(int argc, char** argv) {
     }
 #endif
 
+    // open pid file, obtain file lock and save pid
```
Comment (cambyzju) [inline be/src/service/doris_main.cpp:468]: as we need to move the code location, is it better wrap these codes into a function?

**Comment by GoGoWen** [comment]
I think it is no need to wrap write/read pidfile into a function,  Because
seems no elsewhere in our code to use it.

camby ***@***.***> 于2022年8月25日周四 22:13写道：

**Review by jackwener — DISMISSED** [review]
please rebase to pass CI

**Comment by GoGoWen** [comment]
> 

done

**Comment by hello-stephen** [comment]
when BE is already started, then run "sh start_be.sh --daemon", be.pid file will be overwritten, Do you mean this?
At present, start_be.sh will check if PID process already exists and throw error info if it does.
In some situations, the PID in be.pid file is not the actual PID of doris_be, and if run start_be.sh, the be.pid file will indeed be overwritten， for example， if run this `for i in {1..10};do  bash start_be.sh --daemon & done`, the pid in be.pid is probably wrong，and if run start_be.sh again, the be.pid file will indeed be overwritten. 
So, it looks like this pr didn't really fix the problem.
any other idea?

**Pull request closed without merging**